### PR TITLE
add condition to avoid duplicate expiration calculations

### PR DESCRIPTION
**Pull request by eve-git**
*Issue #, if available:*
https://github.com/bcgov/entity/issues/19788

*Description of changes:*
add condition to avoid duplicate expiration calculations in payment processes for NR reapply

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of the namex license (Apache 2.0).

### CODE REVIEW

File: api/namex/services/name_request/name_request_state.py [hunk 1]
```diff
@@ -76,18 +76,23 @@ def display_reapply_action(nr_model=None) -> Boolean:
...
     try:
         if nr_model and nr_model.stateCd in (State.CONDITIONAL, State.APPROVED):
-            if nr_model.expirationDate and not nr_model.is_expired and not nr_model.has_consumed_name:
-                todays_date = datetime.utcnow().date()
-                expiry_date = nr_model.expirationDate.date()
-
-                delta = expiry_date - todays_date
-                if delta.days <= 14:
-                    return True
+            if not nr_model.is_expired and not nr_model.has_consumed_name:
```
Comment (seeker25) [inline api/namex/services/name_request/name_request_state.py:79]:
```
if nr_model.is_expired or nr_model.has_consumed_name:
    return False
return is_reapplication_eligible(nr_model.expirationDate)
```

File: api/namex/services/name_request/name_request_state.py [hunk 2]
```diff
@@ -76,18 +76,23 @@ def display_reapply_action(nr_model=None) -> Boolean:
...
+                return is_reapplication_eligible(nr_model.expirationDate)
         return False
     except Exception as err:
         raise NameRequestActionError(err)
 
 
+def is_reapplication_eligible(expiration_date) -> Boolean:
+    if expiration_date:
+        todays_date = datetime.utcnow().date()
```
Comment (seeker25) [inline api/namex/services/name_request/name_request_state.py:88]: `datetime.utcnow()` is depreciated

There's an equivalent - `datetime.now(timezone.utc)` I think

**Comment by seeker25** [comment]
Not a namex pro, but thanks for fixing this!

**Review by rarmitag — APPROVED** [review]

**Comment by rarmitag** [comment]
@eve-git - Find out why the tests are failing before merging

**Review by rarmitag — APPROVED** [review]

**Comment by eve-git** [comment]
> @eve-git - Find out why the tests are failing before merging

lots of lint errors all over the code, such as line too long (110>79 characters). I think we need to re-config lint for the coding format since 80 characters/line is too short.
image

File: api/namex/services/name_request/name_request_state.py [hunk 3]
```diff
@@ -76,16 +76,17 @@ def display_reapply_action(nr_model=None) -> Boolean:
...
+            return is_reapplication_eligible(nr_model.expirationDate)
         return False
     except Exception as err:
         raise NameRequestActionError(err)
 
 
 def is_reapplication_eligible(expiration_date) -> Boolean:
     if expiration_date:
-        todays_date = datetime.utcnow().date()
+        todays_date = datetime.now(timezone.utc)
```
Comment (seeker25) [inline api/namex/services/name_request/name_request_state.py:89]: `.date()`

File: api/namex/services/name_request/name_request_state.py [hunk 4]
```diff
@@ -76,18 +76,24 @@ def display_reapply_action(nr_model=None) -> Boolean:
...
+                return False
+            return is_reapplication_eligible(nr_model.expirationDate)
         return False
     except Exception as err:
         raise NameRequestActionError(err)
 
 
+def is_reapplication_eligible(expiration_date) -> Boolean:
+    if expiration_date:
+        todays_date = datetime.now(timezone.utc)
```
Comment (seeker25) [inline api/namex/services/name_request/name_request_state.py:89]: missing `.date()` I think

**Review by seeker25 — APPROVED** [review]

**Pull request merged**